controllers/hosting/model_reconciler.go

```diff
@@ -365,16 +365,16 @@ func (r *modelReconciler) getAndValidateContainerMap(model *commonv1.Model) (map
 // Create a list of containers that a single model requires.
 // This does not include the primary container in the list, because the SageMaker API requires that it be separate.
 // An error is returned if a required container is not defined.
-func (r *modelReconciler) getContainerDefinitions(model *commonv1.Model, containers map[string]*commonv1.ContainerDefinition, primaryContainerHostname string) ([]*commonv1.ContainerDefinition, error) {
+func (r *modelReconciler) getContainerDefinitions(model *commonv1.Model, containers map[string]*commonv1.ContainerDefinition, primaryContainer *commonv1.ContainerDefinition) ([]*commonv1.ContainerDefinition, error) {
 	var modelContainers []*commonv1.ContainerDefinition
 	for _, container := range model.Containers {
-
-		containerName := *(container.ContainerHostname)
 		// Do not store primary container in containers slice.
-		if containerName == primaryContainerHostname {
+		if container == primaryContainer {
```

Comment:
can you provide more detail on how this is solving the problem?
to me, it looks like we are ignoring one of the containers from the list

unless I am missing something, it looks like this would be better solved by putting a check for `len(containers)` where these methods are called

Reply:
oh I see, primaryContainer will be nil in multi container case. Why do we need this check at all